### ISSUE TITLE
Setting DetectionSpeed to noDuplicates prevent the scanner from working after pop and push the view again 

### ISSUE THREAD

**Issue by MiBSchmidt** [juliansteenbakker/mobile_scanner#750]
Hello
to reproduce the error you have to do the following steps:
1. Open the file `barcode_scanner_controller.dart`  from the example app
2. Set the detectionSpeed to noDuplicates
```

   MobileScannerController controller = MobileScannerController(
    torchEnabled: true,
    // formats: [BarcodeFormat.qrCode]
    // facing: CameraFacing.front,
    detectionSpeed: DetectionSpeed.noDuplicates
    // detectionTimeoutMs: 1000,
    // returnImage: false,
  );
````
3. Start the example app
4. Tap on the button "MobileScanner with Controller"
5. Scann a Barcode 
6. Return to main menu
7. Select the button "MobileScanner with Controller"

Now the barcode will not be detected anymore.  If the detectionSpeed is set to normal, everythink works fine.

I guess this issue is related to #619 and #589

I am using a Google Pixel 7 Pro with Android 13 

Greetings 

Michael

**Comment by navaronbracke** [comment]
This is fixed in version 3.5.2

**Comment by Juanchote** [comment]
I am having the same issue with version 3.5.5
Even if I dispose the controller, it doesn't scan the barcodes anymore.

I have Samsung S10+ with android 12.
